components/ai_chat/core/browser/engine/conversation_api_client_unittest.cc

```diff
@@ -1618,6 +1618,295 @@ TEST_F(ConversationAPIUnitTest, ParseResponseEvent_ParsesWebSourcesEvent) {
   EXPECT_FALSE(result3) << "Disallowed favicon host should be filtered out";
 }
 
+TEST_F(ConversationAPIUnitTest,
+       ParseResponseEvent_ParsesWebSourcesEventWithRichResults) {
+  // Test webSources event with valid rich_results data
+  base::Value::Dict event;
+  event.Set("type", "webSources");
+  event.Set("model", "llama-3-8b-instruct");
+
+  // Add regular sources
+  base::Value::List sources;
+  base::Value::Dict source;
+  source.Set("title", "Example Source");
+  source.Set("url", "https://example.com");
+  source.Set("favicon", "https://imgs.search.brave.com/favicon.ico");
+  sources.Append(std::move(source));
+  event.Set("sources", std::move(sources));
+
+  // Add rich_results with nested structure
+  base::Value::List rich_results;
+  base::Value::Dict rich_result_group;
+  base::Value::List results;
+
+  // First rich result item
+  base::Value::Dict rich_item1;
+  rich_item1.Set("type", "knowledge_graph");
+  rich_item1.Set("title", "Knowledge Graph Title");
+  rich_item1.Set("description", "Some description");
+  results.Append(rich_item1.Clone());
+
+  // Second rich result item
+  base::Value::Dict rich_item2;
+  rich_item2.Set("type", "video");
+  rich_item2.Set("url", "https://video.example.com");
+  rich_item2.Set("thumbnail", "https://imgs.search.brave.com/thumb.jpg");
+  results.Append(rich_item2.Clone());
```

Comment:
If you want to make this less verbose, you could consider `base::test::ParseJsonDict(R"({`